test/performance/odp_lock_perf.c

```diff
@@ -0,0 +1,670 @@
+			counter_out[i] = &g->item.all_separate[i].counter;
+		break;
+	}
+}
+
+static int run_test(void *arg)
+{
+	uint64_t nsec;
+	odp_time_t t1, t2;
+	test_thread_ctx_t *thread_ctx = arg;
+	test_global_t *global = thread_ctx->global;
+	test_options_t *test_options = &global->test_options;
+	test_fn_t test_func = thread_ctx->func;
+	volatile uint64_t *counter[MAX_COUNTERS];
+
+	fill_counter_ptrs(global, counter);
+
+	/* Start all workers at the same time */
+	odp_barrier_wait(&global->barrier);
+
+	t1 = odp_time_local();
+	test_func(global, counter, test_options->num_counter);
+	t2 = odp_time_local();
+	nsec = odp_time_diff_ns(t2, t1);
+
+	/* Update stats */
+	thread_ctx->nsec = nsec;
+
+	return 0;
+}
+
+static int start_workers(test_global_t *global, odp_instance_t instance,
+			 test_fn_t func)
+{
+	odph_thread_common_param_t param;
+	int i, ret;
+	test_options_t *test_options = &global->test_options;
+	int num_cpu = test_options->num_cpu;
+	odph_thread_param_t thr_param[num_cpu];
+
+	odph_thread_common_param_init(&param);
+	param.instance = instance;
+	param.cpumask = &global->cpumask;
+
+	for (i = 0; i < num_cpu; i++) {
+		test_thread_ctx_t *thread_ctx = &global->thread_ctx[i];
+
+		thread_ctx->global = global;
+		thread_ctx->idx = i;
+		thread_ctx->func = func;
+
+		odph_thread_param_init(&thr_param[i]);
+		thr_param[i].thr_type = ODP_THREAD_WORKER;
+		thr_param[i].start = run_test;
+		thr_param[i].arg = thread_ctx;
+	}
+
+	ret = odph_thread_create(global->thread_tbl, &param, thr_param,
+				 num_cpu);
+	if (ret != num_cpu) {
+		ODPH_ERR("Failed to create all threads %i\n", ret);
+		return -1;
+	}
+
+	return 0;
+}
+
+static int validate_results(test_global_t *global, validate_fn_t validate)
+{
+	test_options_t *test_options = &global->test_options;
+	volatile uint64_t *counter[MAX_COUNTERS];
+
+	fill_counter_ptrs(global, counter);
+
+	if (validate(global, counter, test_options->num_counter))
+		return -1;
+
+	return 0;
+}
+
+static void print_stat(test_global_t *global)
+{
+	int i, num;
+	double nsec_ave;
+	test_options_t *test_options = &global->test_options;
+	int num_cpu = test_options->num_cpu;
+	uint64_t num_round = (uint64_t)test_options->num_round;
+	uint64_t nsec_sum = 0;
+
+	for (i = 0; i < ODP_THREAD_COUNT_MAX; i++)
+		nsec_sum += global->thread_ctx[i].nsec;
+
+	if (nsec_sum == 0) {
+		printf("No results.\n");
+		return;
+	}
+
+	nsec_ave = nsec_sum / num_cpu;
+	num = 0;
+
+	printf("------------------------------------------------\n");
+	printf("Per thread results (Millions of rounds per sec):\n");
+	printf("------------------------------------------------\n");
+	printf("          1        2        3        4        5        6        7        8        9       10");
+
+	for (i = 0; i < ODP_THREAD_COUNT_MAX; i++) {
+		if (global->thread_ctx[i].nsec) {
+			if ((num % 10) == 0)
+				printf("\n   ");
+
+			printf("%8.3f ",
+			       num_round /
+				       (global->thread_ctx[i].nsec / 1000.0));
```

Comment:
If this didn't fit into single (100 chars) line, it would be more readable to use a temp variable for the calculation and then print the variable.